graphstorm-processing/scripts/script_utils.py

```diff
@@ -0,0 +1,213 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import argparse
+from ast import literal_eval
+from typing import Any, Dict, Sequence, Optional
+
+from botocore.config import Config
+import boto3
+
+
+def check_if_instances_available(instance_type: str, instance_count: int, region: str) -> None:
+    """
+    Checks if we have enough of a quota for the requested instance type and count in a region.
+    Raises a runtime error if the requested capacity exceeds the current quota.
+    """
+    quota_name = f"{instance_type} for processing job usage"
+    quota_value = get_quota_value("sagemaker", quota_name, region)
+
+    if int(quota_value) < instance_count:
+        raise RuntimeError(
+            f"Requested {instance_count=} goes"
+            f" above current ({quota_value=}) for {instance_type=}"
+        )
+
+    # TODO: Use sagemaker_client.list_processing_jobs and
+    # describe_processing_job to get current utilization
+
+
+def get_quota_value(service_code: str, quota_name: str, region: str) -> float:
+    """
+    For a given service code, quota name, and region, return the quota value.
+    """
+
+    def get_value_for_quota_name(
+        quotas: Sequence[Dict[str, Any]], quota_name: str
+    ) -> Optional[float]:
+        for quota in quotas:
+            if quota["QuotaName"] == quota_name:
+                return quota["Value"]
+        return None
+
+    quota_client = boto3.client(
+        "service-quotas",
+        config=Config(
+            region_name=region, connect_timeout=5, read_timeout=60, retries={"max_attempts": 20}
+        ),
+    )
+    quota_paginator_client = quota_client.get_paginator("list_service_quotas")
+
+    quota_response = quota_paginator_client.paginate(
+        ServiceCode=service_code,
+    )
+
+    for quota_set in quota_response:
+        quota_value = get_value_for_quota_name(quota_set["Quotas"], quota_name)
+        if quota_value:
+            break
+
+    if quota_value is None:
+        raise RuntimeError(
+            f"Could not get quota value for {service_code=}, {quota_name=}, and {region=}"
+        )
+    return quota_value
+
+
+def get_max_volume_size_for_processing(region: str) -> int:
```

Comment:
Can we use EFS?

Reply:
This is with respect to the EBS volume that can be attached to the SageMaker instances. I have not looked into EFS for v0.2, but I think that although we can attach EFS for training jobs, it's not clear if it's possible for Processing jobs. Another hurdle would be to get Spark to read/write from EFS possibly.

Reply:
We can look into EFS for v0.3